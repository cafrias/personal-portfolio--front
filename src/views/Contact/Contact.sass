@import '../../_sass/_constants.sass'
@import '../../_sass/_breakpoints.sass'

// ICONS _______________________________________________________________________
=fill-transition
  transition: fill 0.5s

.icon-fill--muted>.to-fill
  fill: $mid-grey
  transition: fill 0.2s

.hover-entry:hover
  &.icon-fill--twitter>.to-fill
    fill: #1da1f2
    +fill-transition
  &.icon-fill--linkedin>.to-fill
    fill: #006699
    +fill-transition
  &.icon-fill--gmail
    fill: #d54b3d
    +fill-transition
  &.icon-fill--fd
    fill: $primary
    +fill-transition
