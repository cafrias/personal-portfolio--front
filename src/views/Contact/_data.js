// @flow

// LOGOS _______________________________________________________________________
import twitterLogo from './_logos/TwitterLogo'
import githubLogo from './_logos/Github'
import linkedInLogo from './_logos/LinkedInLogo'
import gmailLogo from './_logos/GmailLogo'

// TYPES _____________________________________________________________________
import type { FunctionalComponent } from 'react-flow-types'

export type contactEntry = {
  logo: FunctionalComponent<Object>,
  sub: string,
  h: string,
  text: string,
  link: string
}

export const contactInfo = [
  {
    logo: githubLogo,
    sub: 'Code',
    h: 'Github',
    text: 'github.com/cafrias',
    link: 'https://github.com/cafrias'
  },
  {
    logo: twitterLogo,
    sub: 'Social',
    h: 'Twitter',
    text: '@friasdesign',
    link: 'https://twitter.com/friasdesign'
  },
  {
    logo: linkedInLogo,
    sub: 'Social',
    h: 'LinkedIn',
    text: 'linkedin.com/in/cafrias',
    link: 'https://www.linkedin.com/in/cafrias/'
  },
  {
    logo: gmailLogo,
    sub: 'E-mail',
    h: 'Gmail',
    text: 'carlos.a.frias@gmail.com',
    link: 'mailto:carlos.a.frias@gmail.com'
  }
]
