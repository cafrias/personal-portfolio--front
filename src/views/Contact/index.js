// @flow
import React from 'react'

// ASSETS ______________________________________________________________________
import headingImg from './_assets/heading.svg'
import './Contact.sass'

// DATA ________________________________________________________________________
import {contactInfo} from './_data'

// COMPONENTS __________________________________________________________________
import ArticleLine from '../../components/ArticleLine'
import PageTemplate from '../../components/PageTemplate'
import TextArticle from '../../components/TextArticle'

// TYPES _______________________________________________________________________
import type {
  NormalPageProps
} from '../../_types'

const Contact = (props: NormalPageProps) => {
  const {
    ready
  } = props

  return (
    <div className="template-container template-container--last over-x-hidden">
      <TextArticle heading="let's talk!" ready={ready}>
        <p>
          Contact me if you want to propose a project to develop together, hire me,
          {' '}or even talk about programming a little. I am currently
          {' '}<strong>AVAILABLE FOR HIRING</strong>.
        </p>
      </TextArticle>

      <ArticleLine
        className={`fade-side ${
          ready
          ? ''
          : 'fade-right'
        }`}
        heading="Contact Info"
        entries={contactInfo}
        theme="mono"
        hoverEnabled={true}
      />
    </div>
  )
}

export default PageTemplate(Contact, {
  topNavArrowLabel: 'Portfolio',
  last: true,
  headingImg,
  headingAriaLabel: 'Contact',
  nextPage: '',
  previousPage: '/portfolio'
})
