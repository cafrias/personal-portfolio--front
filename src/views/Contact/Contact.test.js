import React from 'react'
import {shallow} from 'enzyme'

import Contact from './index'

describe('Contact', () => {
  const wrapper = shallow(<Contact />)

  it('should render without crashing', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
