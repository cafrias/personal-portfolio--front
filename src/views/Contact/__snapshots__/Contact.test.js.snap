exports[`Contact should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <main
    className="page-template">
    <Connect(Component)
        direction="up"
        label="Portfolio"
        ready={false} />
    <Connect(Filter)
        type="top" />
    <Connect(Filter)
        type="bottom" />
    <header
        className="section-header w-100">
        <PopIn
            onRest={[Function]}>
            <h1
                aria-label="Contact"
                className="section__heading">
                <img
                    alt="About"
                    className="img__heading"
                    src="heading.svg" />
            </h1>
        </PopIn>
    </header>
    <Contact
        ready={false} />
    <div
        className="end-footer">
        Crafted with ❤ by Carlos Frías
    </div>
</main>,
  "nodes": Array [
    <main
      className="page-template">
      <Connect(Component)
            direction="up"
            label="Portfolio"
            ready={false} />
      <Connect(Filter)
            type="top" />
      <Connect(Filter)
            type="bottom" />
      <header
            className="section-header w-100">
            <PopIn
                  onRest={[Function]}>
                  <h1
                        aria-label="Contact"
                        className="section__heading">
                        <img
                              alt="About"
                              className="img__heading"
                              src="heading.svg" />
                  </h1>
            </PopIn>
      </header>
      <Contact
            ready={false} />
      <div
            className="end-footer">
            Crafted with ❤ by Carlos Frías
      </div>
</main>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <PageTemplate />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": PageTemplate {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "titleAnimationEnd": false,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <main
          className="page-template">
          <Connect(Component)
                    direction="up"
                    label="Portfolio"
                    ready={false} />
          <Connect(Filter)
                    type="top" />
          <Connect(Filter)
                    type="bottom" />
          <header
                    className="section-header w-100">
                    <PopIn
                              onRest={[Function]}>
                              <h1
                                        aria-label="Contact"
                                        className="section__heading">
                                        <img
                                                  alt="About"
                                                  className="img__heading"
                                                  src="heading.svg" />
                              </h1>
                    </PopIn>
          </header>
          <Contact
                    ready={false} />
          <div
                    className="end-footer">
                    Crafted with ❤ by Carlos Frías
          </div>
</main>,
        "_debugID": 2,
        "_renderedOutput": <main
          className="page-template">
          <Connect(Component)
                    direction="up"
                    label="Portfolio"
                    ready={false} />
          <Connect(Filter)
                    type="top" />
          <Connect(Filter)
                    type="bottom" />
          <header
                    className="section-header w-100">
                    <PopIn
                              onRest={[Function]}>
                              <h1
                                        aria-label="Contact"
                                        className="section__heading">
                                        <img
                                                  alt="About"
                                                  className="img__heading"
                                                  src="heading.svg" />
                              </h1>
                    </PopIn>
          </header>
          <Contact
                    ready={false} />
          <div
                    className="end-footer">
                    Crafted with ❤ by Carlos Frías
          </div>
</main>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <PageTemplate />,
}
`;
